### PR TITLE
[DNM] Hero Graph remove old block for new block

### LINKED ISSUE #70: [QA] Keep active blocks set to a certain limit
After leaving the block explorer running for a while, the active blocks state never gets purged which renders too many blocks. Either we can have a set upper limit where the graph starts looking wrong and the state action shifts off old blocks when it pushes on new ones once it has hit that limit.
image

### PR DESCRIPTION
**Pull request by schmentle**
### Description
When the explorer is opened, new blocks will keep rolling in and the graph becomes unreadable.
### Screencast
https://share.getcloudapp.com/E0uzYYjj

Closes: tari-project/block-explorer-frontend#70

### CODE REVIEW

**Review by Krakaw — CHANGES_REQUESTED** [review]
Don't modify data in a render. Move this to the redux action. Add an env for `MAX_BLOCKS` and if latestBlocks > MAX_BLOCKS then splice off the last.

**Comment by schmentle** [comment]
> Don't modify data in a render. Move this to the redux action. Add an env for `MAX_BLOCKS` and if latestBlocks > MAX_BLOCKS then splice off the last.

Thanks @Krakaw. I've moved it to the API call.

File: src/config.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ const config = {
         process.env.REACT_APP_EXPLORER_API_DOMAIN + '/ws'
     ),
     tokenName: process.env.REACT_APP_TOKEN_NAME || 'tXTR',
-    initialBlockCount: +(process.env.REACT_APP_INITIAL_BLOCK_COUNT || 100)
+    initialBlockCount: +(process.env.REACT_APP_INITIAL_BLOCK_COUNT || 100),
+    maxBlocks: +(process.env.REACT_APP_MAX_BLOCKS || 100)
```
Comment (Krakaw) [inline src/config.ts:17]: Add the env to the .env.sample file

File: src/helpers/api.ts [hunk 2]
```diff
@@ -25,6 +25,11 @@ export async function fetchBlocksData(limit = 30, sort = 'desc', page = 0): Prom
     if (sort === 'desc') {
         blocks.blocks.sort((a, b) => b.block.header.height - a.block.header.height);
     }
+
+    if(blocks.blocks.length > config.maxBlocks) {
+        blocks.blocks.splice(-1,1);
+    }
```
Comment (Krakaw) [inline src/helpers/api.ts:31]: Move this to the redux action. Also how do we know that we only need to reduce by 1? We could've fetched 200 blocks.

**Review by Krakaw — CHANGES_REQUESTED** [review]

**Comment by delta1** [comment]
@msbodetti any chance you could please take care of @Krakaw's feedback? Then I'll get this approved and merged - it's a definite improvement!

**Comment by schmentle** [comment]
> @msbodetti any chance you could please take care of @Krakaw's feedback? Then I'll get this approved and merged - it's a definite improvement!

Hey @delta1 went off my radar. Let me look at it today! I'll try get something up later today or this week!

**Comment by delta1** [comment]
Awesome thanks! There's no rush 🙏

**Pull request closed without merging**